lib/facter/resolvers/solaris/processors.rb

```diff
@@ -28,6 +28,17 @@ def parse_output(output)
             @fact_list[:physical_count] = output.scan(/chip_id .*/).uniq.size
             @fact_list[:speed] = output.scan(/current_clock_Hz .*/).first.gsub(/[a-zA-z\s]+/, '').to_i
             @fact_list[:models] = output.scan(/brand .*/).map { |elem| elem.gsub(/brand(\s+)/, '') }
+            calculate_threads(output)
+            calculate_cores(output)
+          end
+
+          def calculate_threads(output)
+            @fact_list[:threads_per_core] = output.scan(/module/).size / (output.scan(/core_id .*/).uniq.size -
```

Comment:
Won't `output.scan(/core_id .*/)` also match `pkg_core_id`?

Also, you can use `@fact_list[:logical_count]` instead of calling `output.scan(/module/).size` again. Same for `output.scan(/chip_id .*/).uniq.size`.

Wondering if it might be better to also assign `output.scan(/core_id .*/).uniq.size` to a `@fact_list[:core_count]` variable and use it in these methods.

Reply:
`output.scan(/core_id .*/)` does also match` pkg_core_id`, so i have subtracted from the amount of `core_id` matches the amount of `pkg_core_id` matches, in hopes that this solves the issue